.github/scripts/build_triton_wheel.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+from subprocess import check_call
+from pathlib import Path
+from tempfile import TemporaryDirectory
+import sys
+import shutil
+SCRIPT_DIR = Path(__file__).parent
+
+def read_triton_pin() -> str:
+    with open(SCRIPT_DIR.parent / "ci_commit_pins" / "triton.txt") as f:
+        return f.read().strip()
+
+
+def patch_setup_py(path: Path, *, version: str = "2.0.0", name: str = "triton") -> None:
+    with open(path) as f:
+        orig = f.read()
+    # Replace name
+    orig = orig.replace("name=\"triton\",", f"name=\"{name}\",")
+    # Replace version
+    orig = orig.replace("version=\"2.0.0\",", f"version=\"{version}\",")
```

Comment:
I'd suggest a check to make sure this (and the above) actually replaced something.  If triton upstream changes to `2.1.0` this will silently do the wrong thing.

Reply:
Good idea, added `check_and_replace`